zgw/objecten-api/src/main/kotlin/com/ritense/objectenapi/listener/ZaakObjectListener.kt

```diff
@@ -0,0 +1,135 @@
+                objectTypeGroup.value.forEach {
+                    // For each requestedField update the value in the zaakObject record data
+                    val startPath = it.path.substring(1)
+                    findAndReplaceJsonPath(zaakObject.record.data!!, startPath, extractValue(it.value as JsonNode)!!.toString())
+                }
+
+                // The zaakObject.record.data has now been updated with the new values, update the object in the objecten api
+                updateObject(zaakObject.url, event.documentId, objectTypeGroup.key, zaakObject.record)
+            }
+        }
+    }
+
+    private fun updateObject(objectUrl: URI, documentId: UUID, objectTypeName: String, objectRecord: ObjectRecord) {
+        val objectType = zaakObjectService.getZaakObjectTypes(documentId).first() {
+            it.name == objectTypeName
+        }
+
+        findObjectenApiPlugin(objectUrl)
+            .objectUpdate(
+                objectUrl,
+                ObjectRequest(
+                    objectType.url,
+                    ObjectRecord(
+                        typeVersion = objectRecord.typeVersion,
+                        data = objectRecord.data,
+                        startAt = LocalDate.now(),
+                        correctionFor = objectRecord.index.toString()
+                    )
+                ))
+    }
+
+    /**
+     * Tne json path of the value the application needs to update could be several layers deep like
+     * /profile/adresses/street this method will (recursively) navigate to the value that needs to be updated
+     */
+    private fun findAndReplaceJsonPath(jsonNode: JsonNode, jsonPath: String, newValue: String) {
+        // If the jsonPath no longer contains a '/' we've reached the node we should update
+        if (!jsonPath.contains('/')) {
+            (jsonNode as ObjectNode).put(jsonPath, newValue)
+            return
+        }
+
+        val currentNodePath = jsonPath.substringBefore("/")
+        if (jsonNode.has(currentNodePath)) {
+            val currentNode = jsonNode.get(currentNodePath)
+            val nextPath = jsonPath.substringAfter("/")
+            findAndReplaceJsonPath(currentNode, nextPath, newValue)
+        } else {
+            throw InvalidPropertiesFormatException("Could not find the path $currentNodePath in the json $jsonNode")
+        }
+    }
+
+    private fun findObjectenApiPlugin(objectUrl: URI): ObjectenApiPlugin {
+        val objectenApiPluginInstance = pluginService
+            .createInstance(ObjectenApiPlugin::class.java) { properties: JsonNode ->
+                objectUrl.toString().startsWith(properties.get("url").textValue())
+            }
+
+        requireNotNull(objectenApiPluginInstance) { "No objecten plugin configuration was found for the URL $objectUrl" }
+
+        return objectenApiPluginInstance
+    }
+
+    private fun extractValue(node: JsonNode): Any? {
```

Comment:
Maybe we can move this somewhere else? This is not specific to zaakobjecten, but more of a JSON thing in general.